src/tools/crawler/rpc.rs

```diff
@@ -0,0 +1,49 @@
+use std::{net::SocketAddr, sync::Arc};
+
+use jsonrpsee::http_server::{HttpServerBuilder, RpcModule};
+use parking_lot::Mutex;
+use tracing::debug;
+
+use crate::metrics::NetworkSummary;
+
+pub struct RpcContext(Arc<Mutex<NetworkSummary>>);
+
+impl RpcContext {
+    /// Creates a new RpcContext.
+    pub fn new(known_network: Arc<Mutex<NetworkSummary>>) -> RpcContext {
+        RpcContext(known_network)
+    }
+}
+
+impl std::ops::Deref for RpcContext {
+    type Target = Mutex<NetworkSummary>;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+pub async fn initialize_rpc_server(rpc_addr: SocketAddr, rpc_context: RpcContext) {
+    let server = HttpServerBuilder::default().build(rpc_addr).await.unwrap();
+
+    let module = create_rpc_module(rpc_context);
+
+    debug!("Starting RPC server at {:?}", server.local_addr().unwrap());
+    let _server_handle = server.start(module).unwrap();
+
+    debug!("RPC server was successfully started");
+
+    std::future::pending::<()>().await;
```

Comment:
do we need this? I'd guess it would suffice for the RPC server handle to not go out of scope, which can be achieved if it's kept within the context of `main`